### PR TITLE
Feat blob detection

### PR DESCRIPTION
**Pull request by pauliuskalv**

### CODE REVIEW

File: TOPBanga/VideoFromFile.cs [hunk 1]
```diff
@@ -63,26 +79,54 @@ public void setDeltaText(String text)
...
+                 * Pause the video and ask the user to select the ball
+                 */
+                this.videoTickTimer.Stop();
+                MessageBox.Show("Please select the ball and press Start Detection");
+                this.colorNeeded = true;
+            }
+            Thread.Sleep((int) video.GetCaptureProperty(Emgu.CV.CvEnum.CapProp.Fps));
+        }
+
+        private void skipFrame_Click(object sender, EventArgs e)
```
Comment (mrzilvis) [inline TOPBanga/VideoFromFile.cs]: Can't understand what this does. When tested, seems like doesn't do much.

File: TOPBanga/VideoFromFile.cs [hunk 2]
```diff
@@ -63,26 +79,54 @@ public void setDeltaText(String text)
...
-                else
-                    this.Picture.Image = currentImage.Bitmap;
-            });
-            this.videoTickTimer.Start();
+                    circleFound = true;
+                    break;
+                }
+            }
+            currentImage.Dispose();
+            if (!circleFound)
```
Comment (mrzilvis) [inline TOPBanga/VideoFromFile.cs]: The circle was lost but didn't get the message/video didn't stop

File: TOPBanga/Detection/GameUtil/GoalChecker.cs [hunk 3]
```diff
@@ -6,135 +6,193 @@
...
              */
-            allowedDiff = (zone.topLeft.X + zone.topRight.X) / iterations;
+            int value = 0;
+
+            if (flags.HasFlag(SidePass.Top)) value++;
+            if (flags.HasFlag(SidePass.Bottom)) value++;
+            if (flags.HasFlag(SidePass.Left)) value++;
+            if (flags.HasFlag(SidePass.Right)) value++;
+
+            if (value >= 2) return true;
```
Comment (mrzilvis) [inline TOPBanga/Detection/GameUtil/GoalChecker.cs]: why 2?

File: TOPBanga/VideoFromFile.cs [hunk 4]
```diff
@@ -34,26 +40,36 @@ private void VideoFromFile_Load(object sender, EventArgs e)
...
                 this.currentFrame = this.video.QueryFrame();
+                CvInvoke.Resize(this.currentFrame, this.currentFrame, new Size(Picture.Width, Picture.Height));
                 this.Picture.Image = this.currentFrame.Bitmap;
                 Image<Bgr, byte> currentImage = this.currentFrame.ToImage<Bgr, byte>();
                 this.detector.image = currentImage;
+                this.colorNeeded = true;
             }
         }
 
         private void Picture_Click(object sender, EventArgs e)
```
Comment (mrzilvis) [inline TOPBanga/VideoFromFile.cs]: after choosing color the first time, can't change it anymore.  Seems like user should be able to change the color after the object is lost, but it never gets lost.  I think there should be feature to adjust color anyway

File: TOPBanga/Detection/GameUtil/GoalChecker.cs (hunk 3 above)
Comment (pauliuskalv) [reply TOPBanga/Detection/GameUtil/GoalChecker.cs]: This function will be removed in a pull request, so ignore it for now

File: TOPBanga/VideoFromFile.cs (hunk 4 above)
Comment (pauliuskalv) [reply TOPBanga/VideoFromFile.cs]: This is fixed in the latest commit

File: TOPBanga/VideoFromFile.cs (hunk 2 above)
Comment (pauliuskalv) [reply TOPBanga/VideoFromFile.cs]: This is fixed in the latest commit

File: TOPBanga/VideoFromFile.cs (hunk 1 above)
Comment (pauliuskalv) [reply TOPBanga/VideoFromFile.cs]: This function is meant for the following scenario: if, in the start of the video, the ball is not visible, but later on it does become visible, we use the skipFrame button to advance to the moment where the ball does indeed appear and select it.

**Pull request merged**